tests/rptest/tests/internal_topic_protection_test.py

```diff
@@ -0,0 +1,117 @@
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import subprocess
+import time
+
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.rpk import RpkTool, RpkException
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from rptest.clients.kafka_cat import KafkaCat
+from ducktape.utils.util import wait_until
+
+
+class InternalTopicProtectionTest(RedpandaTest):
+    """
+    Verify that the `kafka_nodelete_topics` and `kafka_noproduce_topics`
+    configuration properties function as intended.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, extra_rp_conf={}, **kwargs)
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.kafka_cat = KafkaCat(self.redpanda)
+        self.rpk = RpkTool(self.redpanda)
+
+    @cluster(num_nodes=3)
+    def kafka_noproduce_topics_test(self):
+        def test_with_produce_fn(produce_fn, failure_exception_type):
+            test_topic = "noproduce_topic"
+            self.kafka_tools.create_topic_with_config(test_topic, 3, 3, {})
+
+            wait_until(lambda: test_topic in self.rpk.list_topics(),
+                       timeout_sec=90,
+                       backoff_sec=3)
+
+            # Ensure topic can't be produced to via the Kafka API when
+            # it's in the kafka_noproduce_topics list.
+            self.rpk.cluster_config_set("kafka_noproduce_topics",
+                                        str([test_topic]))
+
+            try:
+                produce_fn(test_topic, "test_msg")
+            except Exception as e:
+                assert type(e) == failure_exception_type
+            else:
+                assert False, "Call to delete topic returned sucess"
+
+            try:
+                self.rpk.consume(test_topic, n=1, timeout=30)
+            except RpkException:
+                pass
+            else:
+                assert False, "Messages exist in no produce topic"
+
+            # Check that a topic can be removed from the kafka_noproduce_topics
+            # list then produced to.
+            self.rpk.cluster_config_set("kafka_noproduce_topics", str([]))
+            produce_fn(test_topic, "test_msg")
+            self.rpk.consume(test_topic, n=1, timeout=30)
+
+            # Clean up the existing topic
+            self.rpk.delete_topic(test_topic)
+
+            wait_until(lambda: test_topic not in self.rpk.list_topics(),
+                       timeout_sec=90,
+                       backoff_sec=3)
+
+        test_with_produce_fn(
+            lambda topic, msg: self.kafka_cat.produce_one(topic, msg),
+            subprocess.CalledProcessError)
+        test_with_produce_fn(
```

Comment:
nit: the two different clients should be distinct test functions, so that each one runs on a pristine cluster + so that they can run in parallel (same for the other test too).

That can either be two explicit functions, or use `parametrize` on one function.

Reply:
Switching to `parametrize` instead of have internal functions.